### PR TITLE
Improve WC_Customer_Download_Data_Store::get_downloads() performance

### PR DESCRIPTION
**Pull request by rodrigoprimo**
This PR improves WC_Customer_Download_Data_Store::get_downloads() performance by changing the way the SQL query is built. Before this change, this method would get all table fields even when just the permission_id field is returned. Now the method will get from the database only the fields that will be returned. The fields retrieved from the database can be controlled using the parameter `$args['return']`:

- 'objects' (default): all fields are retrieved from the database and an array of WC_Customer_Download objects is returned.
- 'ids': gets and return only permision_ids
- comma separated list of fields (new option added by this commit): gets and return only the required fields. Examples: 'user_email,download_id,order_id' or 'order_id,download_count'.

To give an example, the query executed when `WC_Customer_Download_Data_Store::get_downloads()` is called by `WC_Download_Handler::download_product()` will change from (the only change is in the list fields to retrieve):

```
SELECT * FROM wp_woocommerce_downloadable_product_permissions WHERE 1=1 AND user_email = ? AND order_key = ? AND product_id = ? AND download_id = ? ORDER BY permission_id ASC LIMIT 1
```

To:

```
SELECT permission_id FROM wp_woocommerce_downloadable_product_permissions WHERE 1=1 AND user_email = ? AND order_key = ? AND product_id = ? AND download_id = ? ORDER BY permission_id ASC LIMIT 1
```

On a test database, this simple modification made the query execution time drop from 0.41 seconds to 0.31 seconds.

This PR also fixes all PHPCS violations in all the modified files (commits 6bb79681b95fa16c9912d0bde242429173f2b391 and 16c9bc969483990580ff6285f65f5660f7edd390) and introduces a few improvements to the WC_Tests_Customer_Download test class (commits 9eeb4cd0bddba264f75ad3de822f77a56952dfb5 and 656356530a41c259a188525837ba1658fbfabfa3).

### CODE REVIEW

**Comment by rodrigoprimo** [comment]
@claudiosanches, commit b9f596e was intentionally added to this PR or maybe you meant to add it somewhere else? Asking as it doesn't seem to be related to the original commits in this PR.

**Comment by claudiosanches** [comment]
@rodrigoprimo I made that commit here just to stop packagist.org to send emails to Mike and me. Nothing to worry about.

File: includes/data-stores/class-wc-customer-download-data-store.php [hunk 1]
```diff
@@ -68,38 +72,46 @@ public function create( &$download ) {
...
+	 * @param WC_Customer_Download $download WC_Customer_Download object.
 	 *
-	 * @throws Exception
+	 * @throws Exception Throw exception if invalid download is passed.
 	 */
 	public function read( &$download ) {
 		global $wpdb;
 
 		$download->set_defaults();
+		$raw_download = $wpdb->get_row(
```
Comment (claudiosanches) [inline includes/data-stores/class-wc-customer-download-data-store.php]: Probably better check if ID is valid before trying to do some query.

Comment (rodrigoprimo) [reply includes/data-stores/class-wc-customer-download-data-store.php]: @claudiosanches, on commit 56cb3d2d4f358488a7a7240175a49ecd26bab92b I moved the `! $download->get_id()` check to happen before the query is executed. What I like about this approach is that we don't run a query that is not needed. What I don't like is that I'm duplicating the exception, but I don't see any way around it. Please let me know if you have a suggestion or if it is good as it is now. Thanks.

**Pull request merged**